libgalois/include/katana/PropertyGraph.h

```diff
@@ -155,23 +155,30 @@ class KATANA_EXPORT GraphTopology {
 /// comprise the physical representation of the logical property graph.
 class KATANA_EXPORT PropertyGraph {
 public:
-  /// TypeSetID uniquely identifies/contains a combination/set of types
-  /// TypeSetID is represented using 8 bits
-  /// TypeSetID for nodes is distinct from TypeSetID for edges
-  using TypeSetID = uint8_t;
-  static constexpr TypeSetID kUnknownType = TypeSetID{0};
-  static constexpr TypeSetID kInvalidType =
-      std::numeric_limits<TypeSetID>::max();
-  /// A set of TypeSetIDs
-  using SetOfTypeSetIDs =
-      std::bitset<std::numeric_limits<TypeSetID>::max() + 1>;
-  /// A set of type names
+  /// EntityTypeID uniquely identifies an entity (node or edge) type
+  /// EntityTypeID for nodes is distinct from EntityTypeID for edges
+  /// This type may either be an atomic type or an intersection of atomic types
+  /// EntityTypeID is represented using 8 bits
+  using EntityTypeID = uint8_t;
+  static constexpr EntityTypeID kUnknownType = EntityTypeID{0};
+  static constexpr EntityTypeID kInvalidType =
+      std::numeric_limits<EntityTypeID>::max();
```

Comment:
`numeric_limits` is unsafe. For instance if EntityTypeID were made into an OpaqueID then this would return 0. Use `boost::math::tools::max_value` it will fail if there is no maximum instead of producing terrible bugs.

Reply:
This is the sentinel value, right? Can't we do this when we move this to OpaqueIDs?

Reply:
Yes. OpaqueIDs provide a sentinal, so you could definitely use that here if we change EntityTypeID into an OpaqueID (which I would love to see).